base/strings/search.jl

```diff
@@ -387,27 +452,29 @@ function _rsearchindex(s::String, t::String, i::Integer)
     end
 end
 
-function _rsearchindex(s::ByteArray, t::ByteArray, k::Integer)
-    n = sizeof(t)
-    m = sizeof(s)
+function _rsearchindex(s::AbstractVector{<:Union{Int8,UInt8}}, t::AbstractVector{<:Union{Int8,UInt8}}, k::Integer)
+    n = length(t)
+    m = length(s)
+    f_s = firstindex(s)
+    k < f_s && throw(BoundsError(s, k))
```

Comment:
Same as above: since you are using `_nthbyte` to do 1-based index calculations, you should just shift `k` by `firstindex(s)` and then leave the rest of the calculations unchanged.

i.e. get rid of all the `f_s` changes.